app/src/main/java/org/oppia/android/app/recyclerview/BindableAdapter.kt

```diff
@@ -95,63 +93,15 @@ class BindableAdapter<T : Any> internal constructor(
     internal abstract fun bind(data: T)
   }
 
-  /**
-   * The base builder for [BindableAdapter]. This class should not be used directly--use either
-   * [SingleTypeBuilder] or [MultiTypeBuilder] instead.
-   */
-  abstract class BaseBuilder<BuilderType : Any> {
-    /**
-     * A [WeakReference] to a [LifecycleOwner] for databinding inflation. See [setLifecycleOwner].
-     * Note that this needs to be a weak reference so that long-held references to the adapter do
```

Comment:
This part still seems relevant.

I'm actually wondering if we should keep the ``BaseBuilder``, but just change it so that the lifecycle owner is passed in via the constructor rather than through ``setLifecycleOwner``.

Reply:
Yeah Ben i think we can keep it i noted it is holding a weak reference of the LifecycleOwner. As described to prevent memory leaks.